services/filemanager/s3manager_test.go

```diff
@@ -26,44 +30,17 @@ func TestGetSessionWithAccessKeys(t *testing.T) {
 func TestGetSessionWithIAMRole(t *testing.T) {
 	s3Manager := S3Manager{
 		Config: &S3Config{
-			Bucket:     "someBucket",
+			Bucket: "someBucket",
+			Region: aws.String("someRegion"),
+		},
+		SessionConfig: &awsutils.SessionConfig{
 			IAMRoleARN: "someIAMRole",
 			ExternalID: "someExternalID",
-			Region:     aws.String("someRegion"),
+			Region:     "someRegion",
 		},
 	}
 	awsSession, err := s3Manager.getSession(context.TODO())
 	assert.Nil(t, err)
 	assert.NotNil(t, awsSession)
 	assert.NotNil(t, s3Manager.session)
 }
-
-func TestGetSessionConfigWithAccessKeys(t *testing.T) {
-	s3Manager := S3Manager{
-		Config: &S3Config{
-			Bucket:      "someBucket",
-			AccessKeyID: "someAccessKeyId",
-			AccessKey:   "someSecretAccessKey",
-			Region:      aws.String("someRegion"),
-		},
-	}
-	awsSessionConfig := s3Manager.getSessionConfig()
-	assert.NotNil(t, awsSessionConfig)
-	assert.Equal(t, s3Manager.Config.AccessKey, awsSessionConfig.AccessKey)
-	assert.Equal(t, s3Manager.Config.AccessKeyID, awsSessionConfig.AccessKeyID)
-}
-
-func TestGetSessionConfigWithIAMRole(t *testing.T) {
-	s3Manager := S3Manager{
-		Config: &S3Config{
-			Bucket:     "someBucket",
-			IAMRoleARN: "someIAMRole",
-			ExternalID: "someExternalID",
-			Region:     aws.String("someRegion"),
-		},
-	}
-	awsSessionConfig := s3Manager.getSessionConfig()
-	assert.NotNil(t, awsSessionConfig)
-	assert.Equal(t, s3Manager.Config.IAMRoleARN, awsSessionConfig.IAMRoleARN)
-	assert.Equal(t, s3Manager.Config.ExternalID, awsSessionConfig.ExternalID)
-}
```

Comment:
Aren't these meaningful anymore? Are we introducing any other tests to verify that:

1. what we are fixing is actually fixed 
2. no regression has been introduced in the meantime by these changes

Reply:
Yes `s3Manager.getSessionConfig()` is removed so I have added few more tests to test these functionalities.